packages/server/src/course/course.controller.ts

```diff
@@ -185,4 +190,80 @@ export class CourseController {
     const course = await CourseModel.findOne(courseId);
     await this.icalService.updateCalendarForCourse(course);
   }
+
+  @Get(':id/course_override')
+  @Roles(Role.PROFESSOR)
+  async getCourseOverrides(
+    @Param('id') courseId: number,
+  ): Promise<GetCourseOverridesResponse> {
+    const resp = await UserCourseModel.find({
+      where: { courseId, override: true },
+      relations: ['user'],
+    });
+    return {
+      data: resp.map((row) => ({
+        id: row.id,
+        role: row.role,
+        name: `${row.user.firstName} ${row.user.lastName}`,
+        email: row.user.email,
+      })),
+    };
+  }
+
+  @Post(':id/update_override')
+  @Roles(Role.PROFESSOR)
+  async addOverride(
+    @Param('id') courseId: number,
+    @Body() overrideInfo: UpdateCourseOverrideBody,
+  ): Promise<UpdateCourseOverrideResponse> {
+    const user = await UserModel.findOne({
+      where: { email: overrideInfo.email },
+    });
+    if (!user)
+      throw new BadRequestException(
+        ERROR_MESSAGES.courseController.noUserFound,
+      );
+    const userId = user.id;
+    let userCourse = await UserCourseModel.findOne({
+      where: { courseId, userId },
+    });
+    if (!userCourse) {
+      userCourse = await UserCourseModel.create({
+        userId,
+        courseId,
+        role: overrideInfo.role,
+        override: true,
+      }).save();
+    } else {
+      userCourse.override = true;
+      userCourse.role = overrideInfo.role;
+      await userCourse.save();
+    }
+    return {
+      id: userCourse.id,
+      role: userCourse.role,
+      name: `${user.firstName} ${user.lastName}`,
+      email: user.email,
+    };
+  }
+
+  @Delete(':id/update_override')
+  @Roles(Role.PROFESSOR)
+  async deleteOverride(
+    @Param('id') courseId: number,
+    @Body() overrideInfo: UpdateCourseOverrideBody,
+  ): Promise<void> {
+    const user = await UserModel.findOne({
+      where: { email: overrideInfo.email },
+    });
+    if (!user)
+      throw new BadRequestException(
+        ERROR_MESSAGES.courseController.noUserFound,
+      );
+    const userId = user.id;
+    const userCourse = await UserCourseModel.findOne({
+      where: { courseId, userId },
+    });
```

Comment:
Might want to check that `override: true` in the where statement just to be on the safe side. Although I can't imagine a case right now where a user might have multiple user courses for the same course.